modules/storage-api/src/main/java/org/apache/ignite/internal/storage/DataStorageManager.java

```diff
@@ -39,21 +35,35 @@ public class DataStorageManager implements IgniteComponent {
     /** Mapping: {@link DataStorageModule#name} -> {@link StorageEngine}. */
     private final Map<String, StorageEngine> engines;
 
+    /** Mapping: {@link StorageProfileConfiguration#name()} -> {@link StorageEngine#name()}. */
+    private Map<String, String> profilesToEngines;
+
+    /** Storage configuration. **/
+    private StorageConfiguration storageConfiguration;
+
     /**
      * Constructor.
      *
      * @param engines Storage engines unique by {@link DataStorageModule#name name}.
+     * @param storageConfiguration Storage configuration. Needed to extract the storage profiles configurations after start.
      */
-    public DataStorageManager(Map<String, StorageEngine> engines) {
+    public DataStorageManager(Map<String, StorageEngine> engines, StorageConfiguration storageConfiguration) {
         assert !engines.isEmpty();
 
         this.engines = engines;
+        this.storageConfiguration = storageConfiguration;
     }
 
     @Override
     public CompletableFuture<Void> start() throws StorageException {
         engines.values().forEach(StorageEngine::start);
 
+        profilesToEngines = new HashMap<>();
+
+        storageConfiguration.value().profiles().forEach(profile -> {
```

Comment:
This looks strange, can we simply collect the stream into a map?

Reply:
Yep, fixed